cpp/src/Ice/ConnectionI.cpp

```diff
@@ -2310,6 +2382,19 @@ Ice::ConnectionI::idleCheck(
     // else, nothing to do
 }
 
+void
+Ice::ConnectionI::inactivityCheck() noexcept
+{
+    // Called by the InactivityTimerTask.
+    std::lock_guard lock(_mutex);
+    if (isAtRest())
+    {
+        setState(
+            StateClosing,
+            make_exception_ptr(CloseConnectionException{__FILE__, __LINE__, "connection closed by inactivity timer"}));
```

Comment:
I think it's the correct exception to use even though I find its name odd. I'd rather not create yet another local exception.

Assuming we keep this exception here, we'll need to update its doc comment.